code/processes/gateway.q

```diff
@@ -427,6 +427,11 @@ getserveridstype:{[att;typ]
 
 // execute an asynchronous query
 asyncexecjpts:{[query;servertype;joinfunction;postback;timeout;sync]
+ // Check correct function called
+ if[sync<>.gw.call .z.w;
+  @[neg .z.w;.gw.formatresponse[0b;not sync;"Incorrect function used: ",$[sync;"syncexec";"asyncexec"]];()];
```

Comment:
I think i see how this is working... if the client calls in sync, but calls asyncexec, then this: 

.gw.formatresponse[0b;not sync;"Incorrect function used: ",$[sync;"syncexec";"asyncexec"]]

will signal an error and so it won't get sent via (neg .z.w) back to the client.  Is that correct?  Could you make a comment to that effect if it is?  (it's a little tricky)

Reply:
Yeah that's the idea.
Yeah I'll add that now